### PR TITLE
Rework CAcert

### PR DESCRIPTION
**Pull request by EmilienM**
Remove `CACertSecret`, it's part of the same secret that has clouds.yaml. Hypershift will now create the CA secret based on the data in the cloud credentials.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go
Comment (EmilienM) [inline control-plane-operator/controllers/hostedcontrolplane/cloud/openstack/providerconfig.go:57]: hmm this is wrong. It's not the CA path but the actual CA content. I need to fix this.

**Review by mdbooth — APPROVED** [review]

**Comment by EmilienM** [comment]
@MaysaMacedo when time permits for a review, thanks

**Comment by EmilienM** [comment]
it's not working for me, let's wait until we merge it. I'll ping you again once I resolved it.

**Comment by EmilienM** [comment]
let's merge it as is, we'll need to improve it anyway, with the CLI where we'll provide both the openstack credentials and the cacert if required by the cloud.

**Comment by EmilienM** [comment]
in fact let's wait on this one. I really want to pass the CAcert via the CLI. and then a secret will be created in the namespace; example given for Azure:
https://github.com/shiftstack/hypershift/blob/375c2943704e3565495b3ca4dd756f1c6c6e16c9/cmd/cluster/azure/create.go#L214-L226

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go
Comment (MaysaMacedo) [inline control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:4468]: It might be worth handling the `GetCACertFromCredentialsSecret` in the `ReconcileDeployment` as this is specific to openstack and would maintain the pattern of same function signature for all platforms

**Review by MaysaMacedo — APPROVED** [review]
some idea for the next iteration:

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go
Comment (EmilienM) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:4468]: yeah I know how to do it, and I actually did it at some point but reverted, not sure why. I'll fix it very soon.

**Pull request merged**
